src/cljx/schema/spec/core.cljx

```diff
@@ -0,0 +1,87 @@
+(ns schema.spec.core
+  "Protocol and preliminaries for Schema 'specs', which are a common language
+   for schemas to use to express their structure."
+  (:require
+   #+clj [schema.macros :as macros]
+   [schema.utils :as utils])
+  #+cljs (:require-macros [schema.macros :as macros]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Core spec protocol
+
+(defprotocol CoreSpec
+  "Specs are a common language for Schemas to express their structure.
+   These two use-cases aren't priveledged, just the two that are considered core
+   to being a Spec."
+  (subschemas [this]
+    "List all subschemas")
+  (checker [this params]
+    "Create a function that takes [data], and either returns a walked version of data
+     (by default, usually just data), or a utils/ErrorContainer containing value that looks
+     like the 'bad' parts of data with ValidationErrors at the leaves describing the failures.
+
+     params are: subschema-walker, return-walked?, and cache."))
```

Comment:
should it be `subschema-checker`?

also, maybe clarify what the parameters mean? 

maybe something like:
"params is a map specifying:
subschema-checker - a function for checking subschemas
returned-walked? - a boolean specifying whether to return the original or walked version of the data
cache - a map structure from schema to positive result"